src/main/java/ch/njol/skript/util/Utils.java

```diff
@@ -367,6 +378,105 @@ public static double getBlockHeight(final int type, final byte data) {
+	 * this completable future will complete exceptionally if no players are online.
+	 */
+	public static CompletableFuture<ByteArrayDataInput> sendPluginMessage(String channel, String... data) {
+		return sendPluginMessage(channel, r -> true, data);
+	}
+
+	/**
+	 * Sends a plugin message using the from {@code player}.
+	 *
+	 * The next plugin message to be received through {@code channel} will be assumed to be
+	 * the response.
+	 *
+	 * @param player the player to send the plugin message through
+	 * @param channel the channel for this plugin message
+	 * @param data the data to add to the outgoing message
+	 * @return a completable future for the message of the responding plugin message, if there is one.
+	 * this completable future will complete exceptionally if no players are online.
+	 */
+	public static CompletableFuture<ByteArrayDataInput> sendPluginMessage(Player player, String channel, String... data) {
+		return sendPluginMessage(player, channel, r -> true, data);
+	}
+
+	/**
+	 * Sends a plugin message using the first player from {@link Bukkit#getOnlinePlayers()}.
+	 *
+	 * @param channel the channel for this plugin message
+	 * @param messageVerifier verifies that a plugin message is the response to the sent message
+	 * @param data the data to add to the outgoing message
+	 * @return a completable future for the message of the responding plugin message, if there is one.
+	 * this completable future will complete exceptionally if the player is null.
+	 */
+	public static CompletableFuture<ByteArrayDataInput> sendPluginMessage(String channel,
+			Predicate<ByteArrayDataInput> messageVerifier, String... data) {
+		Player firstPlayer = Iterables.getFirst(Bukkit.getOnlinePlayers(), null);
+		return sendPluginMessage(firstPlayer, channel, messageVerifier, data);
+	}
+
+	/**
+	 * Sends a plugin message.
+	 *
+	 * Example usage using the "GetServers" bungee plugin message channel via an overload:
+	 * <code>
+	 *     Utils.sendPluginMessage("BungeeCord", r -> "GetServers".equals(r.readUTF()), "GetServers")
+	 *     			.thenAccept(response -> Bukkit.broadcastMessage(response.readUTF()) // comma delimited server broadcast
+	 *     			.exceptionally(ex -> {
+	 *     			 	Skript.warning("Failed to get servers because there are no players online");
+	 *     			 	return null;
+	 *     			});
+	 * </code>
+	 *
+	 * @param player the player to send the plugin message through
+	 * @param channel the channel for this plugin message
+	 * @param messageVerifier verifies that a plugin message is the response to the sent message
+	 * @param data the data to add to the outgoing message
+	 * @return a completable future for the message of the responding plugin message, if there is one.
+	 * this completable future will complete exceptionally if the player is null.
+	 */
+	public static CompletableFuture<ByteArrayDataInput> sendPluginMessage(Player player, String channel,

```

Comment:
Please ensure that registering incoming plugin channels each time a plugin message is sent does not cause a memory leak.

Reply:
it shouldn't, registrations (at least for incoming channels) go to a set. I'll double check, though.